cmd/influxd/inspect/dump_tsm/dump_tsm.go

```diff
@@ -0,0 +1,341 @@
+package dump_tsm
+
+import (
+	"encoding/binary"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/influxdata/influxdb/v2/tsdb/engine/tsm1"
+	"github.com/spf13/cobra"
+)
+
+type args struct {
+	dumpIndex  bool
+	dumpBlocks bool
+	dumpAll    bool
+	filterKey  string
+	path       string
+}
+
+func NewDumpTSMCommand() *cobra.Command {
+	var arguments args
+	cmd := &cobra.Command{
+		Use:   "dump-tsm",
+		Short: "Dumps low-level details about tsm1 files",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if arguments.path == "" {
+				cmd.PrintErrf("TSM File not specified\n")
+				return nil
+			}
+			arguments.dumpBlocks = arguments.dumpBlocks || arguments.dumpAll
+			arguments.dumpIndex = arguments.dumpIndex || arguments.dumpAll
+			return dumpTSM(cmd, arguments)
+		},
+	}
+
+	cmd.Flags().StringVar(&arguments.path, "file", "",
+		"Root storage path")
```

Comment:
We expect users to pass a TSM file as this arg, right? If so then I'd suggest:
```suggestion
		"Path to TSM file")
```